pms_l10n_es/models/res_partner.py

```diff
@@ -0,0 +1,27 @@
+import logging
+
+from odoo import api, fields, models
+
+CODE_SPAIN = "ES"
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    ine_code = fields.Char(
+        string="INE State Code",
+        compute="_compute_ine_code",
+        store=True,
+    )
+
+    @api.depends("nationality_id")
+    def _compute_ine_code(self):
+        for record in self:
+            if not record.nationality_id:
+                record.ine_code = False
+            elif record.nationality_id.code != CODE_SPAIN:
+                record.ine_code = record.country_id.code_alpha3
+            else:
```

Comment:
It is necessary to check that the record has country_id, and this field must be in the depends so that when modifying it the ine code is recalculated if necessary

Reply:
ok, done